File: backend/components/metiers/threed/security/core/crypto/crypto.js
// crypto.js – Chiffrement/déchiffrement avancé

const crypto = require('crypto');
const KEY = crypto.randomBytes(32);
const IV = crypto.randomBytes(16);

function encryptData(data) {
  const cipher = crypto.createCipheriv('aes-256-cbc', KEY, IV);
  let encrypted = cipher.update(data, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return encrypted;
}

function decryptData(encrypted) {
  const decipher = crypto.createDecipheriv('aes-256-cbc', KEY, IV);
  let decrypted = decipher.update(encrypted, 'hex', 'utf8');
  decrypted += decipher.final('utf8');
  return decrypted;
}

module.exports = { encryptData, decryptData };

File: backend/components/metiers/threed/security/core/crypto/crypto.py
# crypto.py – Chiffrement/déchiffrement avancé

from cryptography.fernet import Fernet

KEY = Fernet.generate_key()
cipher = Fernet(KEY)


def encrypt_data(data: bytes) -> bytes:
    return cipher.encrypt(data)


def decrypt_data(token: bytes) -> bytes:
    return cipher.decrypt(token)
